amazon-kinesis-client/src/test/java/software/amazon/kinesis/application/TestConsumer.java

```diff
@@ -220,4 +258,53 @@ private void deleteResources(StreamExistenceManager streamExistenceManager, Leas
         log.info("-------------Finished deleting resources.----------------");
     }
 
+    @Data
+    @ToString
+    @AllArgsConstructor
+    public static class StreamScaler implements Runnable {
+        private final KinesisAsyncClient client;
+        private final String streamName;
+        private final Queue<ReshardOptions> scalingFactor;
+        private int totalRotations;
+        private KCLAppConfig testConfig;
+
+        @Override
+        public void run() {
+            try {
+                log.info("----------------------------Starting stream scale----------------------");
+                if (!scalingFactor.isEmpty()) {
```

Comment:
The `else` condition is a one-liner. The `if` forces an indentation of ~23 LoC.

Suggestion: when pragmatic to do so, avoid deep-nesting. Example:
```
if (scalingFactor.isEmpty()) {
   log.info(...);
   return;
}

// rest of code here, unindented one level
```

---
Also, do as much as safely possible before/outside a `try` to avoid the performance impact. Refactoring the `if` (as above) could be done outside the `try` to return faster.